Move SolarInputModel and SolarYieldModel to C++ and remove unnecessary global objects

blammit opened the pull request.

blammit commented:
Moving to draft to add some tests for SolarYieldModel

blammit commented:
Tests added, ready for review.

chriadam commented:
Some testing with the maximal mock config showed a couple of issues:

```
Overview -> Solar Yield -> Fronius Eco -> Product Page
-> qrc:/qt/qml/Victron/VenusOS/components/listitems/ListAcInError.qml:16:2: Unable to assign [undefined] to QString
-> AC totals also has an empty field, which perhaps is not expected?

Overview -> Solar Yield -> Single-phase PV Inverter
-> It shows L1 and L2, but L1 is all empty values.  I assume it should only show L1 (and its values should be the ones which are currently shown as L2)

Overview -> Solar Yield -> Inverter RS Solar 48V -> Product Page -> Overall History
-> The breadcrumbs don't either condense of scroll as expected
```

blammit commented on src/solarinput.cpp:
```diff
@@ -0,0 +1,355 @@
+	qreal singlePhaseCurrent = qQNaN();
+	qreal singlePhaseVoltage = qQNaN();
+
+	for (const AcPhase &phase : m_phases) {
```
Done

blammit commented on src/solarinputmodel.cpp:
```diff
@@ -165,15 +80,234 @@ int SolarInputModel::rowCount(const QModelIndex &) const
+
+	if (device->serviceType() == QStringLiteral("solarcharger")
+			|| device->serviceType() == QStringLiteral("multi")
+			|| device->serviceType() == QStringLiteral("inverter")) {
```
No, as pvinverter doesn't have any trackers, so they are handled in the `else` further down

blammit commented:
> Overview -> Solar Yield -> Fronius Eco -> Product Page
> -> qrc:/qt/qml/Victron/VenusOS/components/listitems/ListAcInError.qml:16:2: Unable to assign [undefined] to QString
> -> AC totals also has an empty field, which perhaps is not expected?

Fixed.

> Overview -> Solar Yield -> Single-phase PV Inverter
> -> It shows L1 and L2, but L1 is all empty values.  I assume it should only show L1 (and its values should be the ones which are > currently shown as L2)

Yes the L1 should be hidden in this case, as L2 is the active phase - there's another task for this, will fix it separately.

> Overview -> Solar Yield -> Inverter RS Solar 48V -> Product Page -> Overall History
> -> The breadcrumbs don't either condense of scroll as expected

Fixed.

chriadam reviewed (approved):
LGTM.  Would be great if Daniel could review also, but I'm also happy for this to be merged as-is to get it more testing etc.
